cmd/homi/setup/cmd.go

```diff
@@ -610,7 +612,33 @@ func Gen(ctx *cli.Context) error {
 		return fmt.Errorf("validators-num(%d) cannot be greater than num(%d)", numValidators, cnNum)
 	}
 
-	privKeys, nodeKeys, nodeAddrs := istcommon.GenerateKeys(cnNum)
+	var (
+		privKeys  []*ecdsa.PrivateKey
+		nodeKeys  []string
+		nodeAddrs []common.Address
+	)
+
+	if len(ctx.String(mnemonic.Name)) == 0 {
+		privKeys, nodeKeys, nodeAddrs = istcommon.GenerateKeys(cnNum)
+	} else {
+		mnemonic := ctx.String(mnemonic.Name)
+		mnemonic = strings.ReplaceAll(mnemonic, ",", " ")
+		// common keys used by web3 tools such as hardhat, foundry, etc.
+		if mnemonic == "test junk" {
+			mnemonic = "test test test test test test test test test test test junk"
+		}
+		path := strings.ToLower(ctx.String(mnemonicPath.Name))
+		if !strings.HasPrefix(path, "m") {
+			switch path {
+			case "klay":
+				path = "m/44'/8217'/0'/0/"
+			default:
+				path = "m/44'/60'/0'/0/"
```

Comment:
I think we can check the value more strictly to detect a typo
```suggestion
			case "eth":
				path = "m/44'/60'/0'/0/"
			default:
				return fmt.Errorf("invalid mnemonic path (format: m/44'/60'/0'/0/)")
```

Reply:
This is better. fixed in 783e404c.